e2e/tests/plugins/notebook/restrictedNotebook.e2e.spec.js

```diff
@@ -0,0 +1,227 @@
+    await page.locator('div.c-ne__text').fill(TEST_TEXT);
+    await page.locator('div.c-ne__text').press('Enter');
+
+    return;
+}
+
+async function dragAndDropEmbed(page) {
+    // Click button:has-text("Create")
+    await page.locator('button:has-text("Create")').click();
+    // Click li:has-text("Sine Wave Generator")
+    await page.locator('li:has-text("Sine Wave Generator")').click();
+    // Click form[name="mctForm"] >> text=My Items
+    await page.locator('form[name="mctForm"] >> text=My Items').click();
+    // Click text=OK
+    await page.locator('text=OK').click();
+    // Click text=Open MCT My Items >> span >> nth=3
+    await page.locator('text=Open MCT My Items >> span').nth(3).click();
+    // Click text=Unnamed CUSTOM_NAME
+    await Promise.all([
+        page.waitForNavigation(),
+        page.locator('text=Unnamed CUSTOM_NAME').click()
+    ]);
+
+    await page.dragAndDrop(SINE_WAVE_GENERATOR, DROP_AREA);
+
+    return;
+}
+
+async function lockPage(page) {
+    const commitButton = page.locator(COMMIT_BUTTON_TEXT);
+    await commitButton.click();
+
+    // confirmation dialog click
+    await page.locator('text=Lock Page').click();
+
+    return;
+}
+
+async function openContextMenuRestrictedNotebook(page) {
+    // Click text=Open MCT My Items >> span >> nth=3
+    await page.locator('text=Open MCT My Items >> span').nth(3).click();
+
+    // Click a:has-text("Unnamed CUSTOM_NAME")
+    await page.locator(`a:has-text("Unnamed ${CUSTOM_NAME}")`).click({
+        button: 'right'
+    });
+
+    return;
+}
+
+test.describe('Restricted Notebook', () => {
+
+    test.beforeEach(async ({ page }) => {
+        // eslint-disable-next-line no-undef
+        await page.addInitScript({ path: path.join(__dirname, 'addRestrictedNotebook.js') });
+
+        page.on('console', msg => console.log(msg.text()));
+
+        //Go to baseURL
+        await page.goto('/', { waitUntil: 'networkidle' });
+
+        //Click the Create button
+        await page.click('button:has-text("Create")');
+
+        // Click text=Example Imagery
+        await page.click(`text=${CUSTOM_NAME}`); // this inherently tests renamability
+
+        // Click text=OK
+        await Promise.all([
+            page.waitForNavigation({waitUntil: 'networkidle'}),
+            page.click('text=OK'),
+            //Wait for Save Banner to appear
+            page.waitForSelector('.c-message-banner__message')
+        ]);
+        //Wait until Save Banner is gone
+        await page.waitForSelector('.c-message-banner__message', { state: 'detached'});
+        await expect.soft(page.locator('.l-browse-bar__object-name')).toContainText(`Unnamed ${CUSTOM_NAME}`);
+    });
+
+    test('Can be deleted if there are no locked pages', async ({ page }) => {
+        await openContextMenuRestrictedNotebook(page);
+
+        const menuOptions = page.locator('.c-menu ul');
+
+        await expect.soft(menuOptions).toContainText('Remove');
```

Comment:
We should remove the object and verify that it is removed from whatever-context-this-is-removing it from

OR

Rename the test to 'Shows the remove button from Context Menu'

Reply:
Done and done, removed it and checked that it's gone.